lib/CompilerSwiftSyntax/CMakeLists.txt

```diff
@@ -8,6 +8,10 @@ endif()
 
 # Build swift-syntax libraries with FetchContent.
 function(includeSwiftSyntax)
+  set(CMAKE_SYSTEM_PROCESSOR ${SWIFT_HOST_VARIANT_ARCH})
+  set(CMAKE_OSX_ARCHITECTURES ${SWIFT_HOST_VARIANT_ARCH})
```

Comment:
This may trip up Swift since CMake doesn't support using `CMAKE_OSX_ARCHITECTURES` with Swift.

Reply:
With my local test not setting `CMAKE_OSX_ARCHITECTURES` resulted incorrect `-arch` in C compiler arguments, so this seems to be needed.